### DISCUSSION TITLE
[to Dev] Any recommended tools you trust for making well formatted pdfs?

### DISCUSSION THREAD

**Discussion by TFWol** [gildas-lormeau/SingleFile#1561] [Q&A] [answered]
Since you're working with the ins and outs of saving webpages so often, I was wondering if there were tools you'd recommend to make correctly formatted pdfs.

For example, when dealing with pages like [this](https://lintalist.github.io/), if you try saving to pdf via print, it pretty much gets ruined.

There's pandoc, but it's difficult to produce consistently good outputs.

As for browser addons, there's quite a few, but it's hard for me to determine which are trustworthy (not shoving sneaky things in like telemetry, etc.).

So I'm curious what _you_ trust (btw, I see the list of projects on the main readme, but are those something you use?)

**Comment by gildas-lormeau** [comment, marked as answer]
To be honest, from a purely personal point of view, I don't think saving a web page in PDF format is a suitable solution. Firstly, it can be complicated to keep a suitable layout in paginated A4 format. Secondly, the PDF format is very “destructive” from a semantic point of view. For example, the notion of paragraph doesn't exist in PDF format, which is very unfortunate when archiving text content. Finally, the PDF format is extremely complicated, and I don't find it suitable for this use case. For these reasons, I don't know of any tools for obtaining well-formed PDF files for web pages.

I do not use the list of tools listed in the README. Normally, these are projects you can trust though.

**Reply by TFWol** [reply]
Thank you very much for taking the time to reply.

I really dislike using PDFs myself. 
Got to for now though; tutorials are the main thing I save and Obsidian conveniently renders them. They're also very compatible for AI projects where I can toss in a manual and ask it questions.

I've wanted to migrate to Markdown, but haven't found a good workflow yet.
